src/dispatch/signal/views.py

```diff
@@ -104,12 +105,24 @@ def create_signal_instance(
             detail=[{"msg": msg}],
         ) from None
 
-    signal_instance = signal_service.create_instance(
-        db_session=db_session, signal_instance_in=signal_instance_in
-    )
-    signal_instance.signal = signal
-    db_session.commit()
-    return signal_instance
+    try:
+        signal_instance = signal_service.create_instance(
+            db_session=db_session, signal_instance_in=signal_instance_in
+        )
+    except IntegrityError as e:
+        if type(e.orig) is UniqueViolation:
+            msg = f"The id: {signal_instance_in.id} already exists in the database. Please provide a unique id."
```

Comment:
```suggestion
            msg = f"Signal instance id {signal_instance_in.id} already exists in the database. Please provide a unique id."
```

Reply:
Will the info in this message be enough to troubleshoot on the other side?

Reply:
Probably, it could be improved though. I'll add some additional information.